### PR TITLE
Initial implementation of packet shredder

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
Need a mechanism to split multiple entries into MTU sized UDP packets

#### Summary of Changes
Added a new utility to split packets into chunks

#5294

### CODE REVIEW

File: core/src/shred.rs [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+        self.finalize_shred(current_shred);
+        Ok(())
+    }
+}
+
+impl Shredder {
+    pub fn new(slot: u64, parent: Option<u64>, fec_ratio: u64, signer: &Arc<Keypair>) -> Self {
+        Shredder {
+            slot,
+            index: 0xff,
```
Comment (sagar-solana) [inline core/src/shred.rs]: set to 0

File: core/src/shred.rs [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+            parent,
+            _fec_ratio: fec_ratio,
+            signer: signer.clone(),
+            shreds: vec![],
```
Comment (sagar-solana) [inline core/src/shred.rs]: ..Shredder::default()

File: core/src/shred.rs [hunk 3]
```diff
@@ -0,0 +1,426 @@
...
+}
+
+impl Shredder {
+    pub fn new(slot: u64, parent: Option<u64>, fec_ratio: u64, signer: &Arc<Keypair>) -> Self {
+        Shredder {
+            slot,
+            index: 0xff,
+            parent,
+            _fec_ratio: fec_ratio,
+            signer: signer.clone(),
```
Comment (sagar-solana) [inline core/src/shred.rs]: see if you can remove this clone and the ones from `sign` below

Comment (pgarg66) [reply core/src/shred.rs]: Removed from `sign()`

File: core/src/shred.rs [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+}
+
+#[derive(Default)]
+pub struct Shredder {
+    slot: u64,
+    index: u32,
+    parent: Option<u64>,
+    _fec_ratio: u64,
+    signer: Arc<Keypair>,
+    pub shreds: Shreds,
```
Comment (rob-solana) [inline core/src/shred.rs]: I'm not a fan of this kind of aliasing, Vec<Shred> is preferable to me

**Pull request merged**